test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorResolverTest.cs

```diff
@@ -41,6 +42,465 @@ private static TagHelperDescriptor Valid_InheritedTagHelperDescriptor
             }
         }
 
+
+        public static TheoryData ResolveDirectiveDescriptorsInvalidTagHelperPrefixData
+        {
+            get
+            {
+                var assemblyA = AssemblyName;
+                var stringType = typeof(string);
+                var assemblyB = stringType.GetTypeInfo().Assembly.GetName().Name;
+                var defaultAssemblyLookups = new Dictionary<string, IEnumerable<Type>>
+                {
+                    { assemblyA, new[] { Valid_PlainTagHelperType, Valid_InheritedTagHelperType } },
+                    { assemblyB, new[] { stringType } }
+                };
+                var directiveLocation1 = new SourceLocation(1, 2, 3);
+                var directiveLocation2 = new SourceLocation(4, 5, 6);
+                var multipleDirectiveError =
+                    "Invalid tag helper directive '{0}'. Cannot have multiple '{0}' directives on a page.";
+                var invalidTagHelperPrefixValueError =
+                    "Invalid tag helper directive '{0}' value. '{1} is not allowed in prefix '{2}'.";
+
+                return new TheoryData<Dictionary<string, IEnumerable<Type>>, // descriptorAssemblyLookups
+                                      IEnumerable<TagHelperDirectiveDescriptor>, // directiveDescriptors
+                                      IEnumerable<TagHelperDescriptor>, // expectedDescriptors
+                                      IEnumerable<RazorError>> // expectedErrors
+                {
+                    {
+                        defaultAssemblyLookups,
+                        new[]
+                        {
+                            new TagHelperDirectiveDescriptor(
+                                "th:",
```

Comment:
test other valid prefixes such as "th-" or just "th".  current coverage makes it seem like the colon is required.